src/geoengine/datasets.md

```diff
@@ -0,0 +1,19 @@
+# Datasets
+
+A dataset is a loadable unit in Geo Engine.
+It is a parameter of a source operator (e.g., a [`GdalSource`](../operators/gdalsource.md)) and identifies the data that is loaded.
+Geo Engine supports different types of datasets, reflected by a `DataId`, which are internal and external data.
+
+# Internal dataset
+
+An internal dataset is a dataset that is stored in the Geo Engine.
+Thus, it is efficiently accessible and can be used in workflows.
+The dataset is identified by a `DatasetId` and contains a `DatasetDefinition` that describes the data.
+The `DatasetId` is usually a UUID.
+
+# External dataset
+
+An external dataset is a dataset that is not stored in the Geo Engine.
+Geo Engine accesses it from a foreign location.
+The dataset is identified by an `ExternalDataId` that consists of a `DatasetProviderId` and a `LayerId`.
```

Comment:
it's "DataProvider" and not "DatasetProvider"

Reply:
[4e711e5](https://github.com/geo-engine/docs/pull/44/commits/4e711e5fb3b6454093eefc44893174e830565a99)